main.bal

```diff
@@ -1,56 +1,101 @@
+        error? e = self.handleInput(caller, data);
+
+        if e is error {
+            return;
+        }
+    }
+
+    byte[] buffer = [];
+    int? expectedLen = ();
+    string messageContent = "";
+    int n = 0;
+    string[] array = ["Content-Length: 171\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":{\"capabilities\":{\"textDocumentSync\":2,\"completionProvider\":{\"resolveProvider\":true},\"workspace\":{\"workspaceFolders\":{\"supported\":true}}}}}", 
+    "Content-Length: 201\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":0,\"method\":\"client/registerCapability\",\"params\":{\"registrations\":[{\"id\":\"9716545a-b5e9-4b32-94e1-be674a62eb04\",\"method\":\"workspace/didChangeConfiguration\",\"registerOptions\":{}}]}}", 
+    "Content-Length: 204\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":1,\"method\":\"client/registerCapability\",\"params\":{\"registrations\":[{\"id\":\"36b409a1-be16-4010-b7bf-a26b6f659596\",\"method\":\"workspace/didChangeWorkspaceFolders\",\"registerOptions\":{}}]}}", 
+    "Content-Length: 177\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":2,\"method\":\"workspace/configuration\",\"params\":{\"items\":[{\"scopeUri\":\"file:///c%3A/Users/Mindula/Desktop/vext/out.txt\",\"section\":\"languageServerExample\"}]}}", 
+    "Content-Length: 144\r\n\r\n{\"jsonrpc\":\"2.0\",\"method\":\"textDocument/publishDiagnostics\",\"params\":{\"uri\":\"file:///c%3A/Users/Mindula/Desktop/vext/out.txt\",\"diagnostics\":[]}}"];
+
+    function handleInput(tcp:Caller caller, readonly & byte[] data) returns error? {
+
+        if self.expectedLen == () {
+
+            self.buffer = data;
+            self.expectedLen = check extractContentLength(data);
 
-        string|error firststring = string:fromBytes(data);   
+        } else {
 
-            if firststring is string{
-                string finalstring = regex:replaceAll(firststring, "/[^ -~]+/g", "");
-                int x = <int> finalstring.indexOf("{");
-                int y = finalstring.length();
-                string secondstring = finalstring.substring(x,y);
-                
-                // io:println(secondstring);
+            self.buffer.push(...data);
+            self.messageContent = self.contentLengthRemover(self.buffer);
+            if (self.expectedLen == self.messageContent.length()) {
+                json jsonObject = self.messageContent.toJson();
+                io:println(jsonObject);
 
-                io:Error? fileWriteString2 = io:fileWriteString("file7.json",secondstring,io:APPEND);
+                byte[] byteMessage = self.array[self.n].toBytes();
+                check caller->writeBytes(byteMessage);
+                self.n += 1;
 
-                // json|error j = secondstring.fromJsonString();
-                // io:println(j);
+                self.expectedLen = ();
+                self.messageContent = "";
+                self.buffer.removeAll();
 
+            } else {
+                io:println("Content Length Does not Match the Message Length");
```

Comment:
I think we don't need to print this, it's OK for the massage to less than expectedLen even on the second message, that means there will be a third part ect.

But we should check following:
```suggestion
            }
            else if self.expectedLen < self.messageContent.length() {
                // panic with an error
            }
```
because in this case we have gotten too long message and something has gone wrong.